test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -290,46 +406,69 @@ private void CopyRestoreArtifacts(string artifactsDirectory, string pathToSdkInC
-
-                IEnumerable<DirectoryInfo> frameworkArtifactFolders = projectArtifactsFolder.EnumerateDirectories().Where(folder => folder.FullName.Contains("netcoreapp5.0"));
+                var projectArtifactsBinFolder = Path.Combine(artifactsDirectory, projectName, toolsetVersion, "bin", configuration);
 
-                if (!frameworkArtifactFolders.Any())
-                {
-                    frameworkArtifactFolders = projectArtifactsFolder.EnumerateDirectories().Where(folder => folder.FullName.Contains("netstandard2.0"));
-                }
+                var tfmToCopy = GetTfmToCopy(sdkTfm, projectArtifactsBinFolder);
+                var frameworkArtifactsFolder = new DirectoryInfo(Path.Combine(projectArtifactsBinFolder, tfmToCopy));
 
-                foreach (var frameworkArtifactsFolder in frameworkArtifactFolders)
+                var fileName = projectName + ".dll";
+                File.Copy(
+                        sourceFileName: Path.Combine(frameworkArtifactsFolder.FullName, fileName),
+                        destFileName: Path.Combine(pathToSdkInCli, fileName),
+                        overwrite: true);
+                // Copy the restore targets.
+                if (projectName.Equals(restoreProjectName))
                 {
-                    var fileName = projectName + ".dll";
                     File.Copy(
-                            sourceFileName: Path.Combine(frameworkArtifactsFolder.FullName, fileName),
-                            destFileName: Path.Combine(pathToSdkInCli, fileName),
-                            overwrite: true);
-                    // Copy the restore targets.
-                    if (projectName.Equals(restoreProjectName))
-                    {
-                        File.Copy(
-                            sourceFileName: Path.Combine(frameworkArtifactsFolder.FullName, restoreTargetsName),
-                            destFileName: Path.Combine(pathToSdkInCli, restoreTargetsName),
-                            overwrite: true);
-                    }
+                        sourceFileName: Path.Combine(frameworkArtifactsFolder.FullName, restoreTargetsName),
+                        destFileName: Path.Combine(pathToSdkInCli, restoreTargetsName),
+                        overwrite: true);
                 }
             }
         }
 
+        private string GetTfmToCopy(NuGetFramework sdkTfm, string projectArtifactsBinFolder)
+        {
+            var compiledTfms =
+                Directory.GetDirectories(projectArtifactsBinFolder) // get all directories in bin folder
+                .Select(Path.GetFileName) // just the folder name (tfm)
+                .ToDictionary(folder => NuGetFramework.Parse(folder));
+
+            var reducer = new FrameworkReducer();
+            var selectedTfm = reducer.GetNearest(sdkTfm, compiledTfms.Keys);
```

Comment:
Good to know that we can use `GetNearest`  to find the right bin/Debug/TFM folder, to patch the dotnet. So we can get rid of hardcoded TFM and the if statement in selecting TFM folder.

Reply:
We have multiple methods called `GetNearest` in several types, and I have no idea what the difference is between them, or if I'm using the "right" one, but it seems to work.